### PR TITLE
Jormungandr: Add free_radius parameters

### PR DESCRIPTION
**Pull request by mbounabi**
Tickets JIRA:
https://jira.canaltp.fr/browse/NAVP-820
https://jira.canaltp.fr/browse/NAVP-821
https://jira.canaltp.fr/browse/NAVP-822

### CODE REVIEW

File: documentation/slate/source/includes/apis.md [hunk 1]
```diff
@@ -1006,6 +1008,18 @@ Examples:
 * A user doesn't like line A metro in hers city. She adds the parameter `forbidden_uris[]=line:A` when calling the API.
 * A user would only like to use Buses and Tramways. She adds the parameter `allowed_id[]=physical_mode:Bus&allowed_id[]=physical_mode:Tramway`.
 
+### Precisions on `free_radius_from/free_radius_to`
+
+These parameters allow to find the nearest stop point (within free_radius distance) to the given coordinates.
```
Comment (pbougue) [inline documentation/slate/source/includes/apis.md]: I'd put it like this (remove "allow to", add "Then it" + "skipping" + "those nearest stop points"):
```
These parameters find the nearest stop point (within free_radius distance) to the given coordinates.
Then it allows skipping walking sections between the point of departure/arrival and those nearest stop points.
```

**Review by pbougue — COMMENTED** [review]
looks good!

**Review by benoit-bst — APPROVED** [review]

**Review by nonifier — APPROVED** [review]
All good to me =)

File: source/jormungandr/tests/journey_common_tests.py [hunk 2]
```diff
@@ -853,6 +853,20 @@ def test_journey_from_non_valid_object_type(self):
...
+        r = self.query('/v1/coverage/main_routing_test/journeys?from=coord%3A8.98311981954709e-05%3A8.98311981954709e-05&to=stopA&datetime=20120614080000')
+
+        assert(r['journeys'][0]['sections'][0]['type'] == 'street_network')
+        assert (r['journeys'][0]['sections'][0]['duration'] != 0)
+
+        # Query journeys with free radius
+        r = self.query('/v1/coverage/main_routing_test/journeys?from=coord%3A8.98311981954709e-05%3A8.98311981954709e-05&to=stopA&datetime=20120614080000&free_radius_from=200')
+
+        assert(r['journeys'][0]['sections'][0]['type'] == 'crow_fly')
+        assert (r['journeys'][0]['sections'][0]['duration'] == 0)
```
Comment (TeXitoi) [inline source/jormungandr/tests/journey_common_tests.py]: Please add a test for `free_radius_to`

**Comment by benoit-bst** [comment]
For QA test, in JIRA ticket (https://jira.canaltp.fr/browse/NAVP-819)

File: source/jormungandr/jormungandr/scenarios/helper_classes/fallback_durations.py [hunk 3]
```diff
@@ -95,10 +95,23 @@ def _do_request(self):
...
+        free_access = self._places_free_access.wait_and_get()
+
+        free_radius_distance = None
+        if self._direct_path_type == StreetNetworkPathType.BEGINNING_FALLBACK:
+            free_radius_distance = self._request.free_radius_from
+        elif self._direct_path_type == StreetNetworkPathType.ENDING_FALLBACK:
+            free_radius_distance = self._request.free_radius_to
+
+        if free_radius_distance is not None:
+            for p in proximities_by_crowfly:
```
Comment (xlqian) [inline source/jormungandr/jormungandr/scenarios/helper_classes/fallback_durations.py]: this should run (much) faster

```python
try:
    # Python 2
    from future_builtins import filter # returns a iterator, not a list
except ImportError:
    # Python 3
    pass


free_access.free_radius.update(filter(lambda p : p.distance < free_radius_distance, proximities_by_crowfly))

# or 
free_access.free_radius.update((p for p  in  proximities_by_crowfly if p.distance < free_radius_distance))

```

**Review by TeXitoi — APPROVED** [review]

**Comment by pbougue** [comment]
Jenkins, test this please

**Comment by mbounabi** [comment]
Jenkins, test this please

File: source/jormungandr/jormungandr/scenarios/helper_classes/fallback_durations.py (hunk 3 above)
Comment (mbounabi) [reply source/jormungandr/jormungandr/scenarios/helper_classes/fallback_durations.py]: Done... With 'p.uri' and 1 parenthesis only

**Comment by mbounabi** [comment]
Jenkins, test this please

**Comment by mbounabi** [comment]
Jenkins, test this please

**Review by TeXitoi — APPROVED** [review]

**Pull request merged**